service/src/main/frontend/README.md

```diff
@@ -0,0 +1,87 @@
+    git submodule foreach 'git checkout main || :'
+
+    # Then build the datawave project and produce images for each microservice:
+
+    mvn -Pcompose -Dmicroservice-docker -Dquickstart-docker -Ddeploy -Dtar -Ddist clean install -DskipTests -Dmaven.build.cache.enabled=false
+
+    # Spin up all the containers for the microservice after full datawave build:
+
+    cd /datawave/docker
+    docker compose up -d
+    ```
+
+3. Now if you go to https://localhost:8643/dictionary/data/v2/#/ you should be able to see the Data Dictionary Version 2 and if you go to https://localhost:8643/dictionary/data/v1/ you will be able to see the predecessor.
+
+##### To Make a Quick Change
+1. If you would like to make a quick change either to the backend or frontend of the service, run the following:
+
+    ```
+    # First go to the docker directory and take down the containers:
+
+    cd /datawave/docker
+    docker compose down
+
+    # Enter only the Dictionary Service Directory and rebuild the jar and image:
+
+    cd /datawave/microservices/services/dictionary/
+
+    mvn -Pcompose -Dmicroservice-docker -Dquickstart-docker -Ddeploy -Dtar -Ddist clean install -DskipTests -Dmaven.build.cache.enabled=false
+
+    # Go back to the docker directory and spin up the containers again.
+    cd /datawave/docker
+    docker compose up -d
+    ```
+2. Then you should be able to hit this endpoint again: https://localhost:8643/dictionary/data/v2/#/
+
+### Instructions for Quasar Dev Mode (Uses Vite.js)
+1. To run this application in DEV mode, first follow the steps above to build and run the containers. This process only needs to be completed once to retrieve the necessary endpoints.
+2. Next, in order to run in DEV mode, you will first need to install [Node.js and NPM](https://nodejs.org/en/download/package-manager/). Ensure you installed them by running `npm -v` and `node -v` to check their versions.
+3. Then you will need to globally install the the Quasar/CLI package. This can be done using npm by running:
+
+    ```
+    npm i -g @quasar/cli
+    ```
+4. Once installed, you can make quick changes in Quasar's Dev Mode. Open your code editor (such as VSCode) and ensure you have the official Vue and Vite extensions installed. Then, open the Datawave project directory in your code editor. Navigate to the main repository of the Data Dictionary service in your code editor and in your terminal which is:
+
+    ```
+    cd /datawave/microservices/services/dictionary/service/src/main/frontend
+    ```
+5. Now, make a change to the `DataDictionary.vue` file (where most of the frontend components are housed) or any TypeScript file, such as logging a message to the terminal or modifying the color of a button. While still in this directory, execute the following command:
+
+    ```
+    quasar dev
+    ```
+*For more information, please refer to [Quasar's documentation](https://quasar.dev/start/quasar-cli#running-without-the-global-quasar-cli) on running in development mode.*
```

Comment:
Could we do this in a container instead of requiring nodejs and quasar to be installed on the dev machine? If you prefer we could work that as a separate ticket and move on from that in this PR.

Reply:
Yeah I can add that is a ticket. Do you suggest including that in the Epic?

Reply:
Yeah that's a good idea.